liblwgeom/lwgeom_median.c

```diff
@@ -158,12 +183,12 @@ lwmpoint_extract_points_3d(const LWMPOINT* g, uint32_t* ngeoms)
 LWPOINT*
 lwmpoint_median(const LWMPOINT* g, double tol, uint32_t max_iter, char fail_if_not_converged)
 {
-	uint32_t npoints; /* we need to count this ourselves so we can exclude empties */
-	uint32_t i;
+	size_t npoints; /* we need to count this ourselves so we can exclude empties and weightless points */
+	size_t i;
 	double delta = DBL_MAX;
 	double* distances;
-	POINT3D* points = lwmpoint_extract_points_3d(g, &npoints);
-	POINT3D median;
+	POINT4D* points = lwmpoint_extract_points_4d(g, &npoints);
+	POINT4D median;
```

Comment:
Does this need to be 4D? The returned values are either 2D or 3D.

Reply:
Ansi C does not allow declaring variables after code (my compiler complains about this)